File: 66. Plus One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        for i in range(len(digits)-1, -1, -1):
            carry, digits[i] = divmod(digits[i]+1, 10)
            print(carry)
            print(digits[i])
            if not carry:
                return digits
        digits.insert(0,1)
        return digits

File: 66. Plus One.java
class Solution {
    public int[] plusOne(int[] digits) {
        if (digits==null || digits.length==0) {
            return digits;
        }
        int n = digits.length;
        int carry = digits[n-1] + 1;
        digits[n-1] = carry % 10;
        carry = carry / 10;
        int idx = n-2;
        while (idx >=0 && carry != 0) {
            carry = digits[idx] + carry;
            digits[idx] = carry % 10;
            carry /= 10;
            idx--;
        }
        if (carry != 0) {
            int[] ans = new int[n+1];
            ans[0] = carry;
            for (int i=1; i<n+1; i++) {
                ans[i] = digits[i-1];
            }
            return ans;
        } else {
            return digits;
        }
     }
}